### PR TITLE
pkgs/stdenv/linux: Add powerpc64 bootstrap-files

### PR DESCRIPTION
**Pull request by OPNA2608**
## Description of changes

`powerpc64-linux` tracking issue: https://github.com/NixOS/nixpkgs/issues/253715
Previous attempt: https://github.com/NixOS/nixpkgs/pull/177001, CC @minijackson
bootstrap files need to be put onto tarballs.nixos.org, CC @lovesegfault for uploading

---

sha256sum of files to be uploaded:

```
$ sha256sum /nix/store/8frm8kk8gzpv31r289ai5jgkwfikmpm4-stdenv-bootstrap-tools-powerpc64-unknown-linux-gnuabielfv2/on-server/*
0af311476b54b399f3024b92e9c518363acd2b15f713e83d0bb4fb3f8f26d98b  /nix/store/8frm8kk8gzpv31r289ai5jgkwfikmpm4-stdenv-bootstrap-tools-powerpc64-unknown-linux-gnuabielfv2/on-server/bootstrap-tools.tar.xz
529a053a8022e89357aaa608e57aaddaa7c3ded93d633916ddca92bb81e22125  /nix/store/8frm8kk8gzpv31r289ai5jgkwfikmpm4-stdenv-bootstrap-tools-powerpc64-unknown-linux-gnuabielfv2/on-server/busybox
```

Suggested commands to upload files to 'tarballs.nixos.org':

```
    $ nix-store --realize /nix/store/8frm8kk8gzpv31r289ai5jgkwfikmpm4-stdenv-bootstrap-tools-powerpc64-unknown-linux-gnuabielfv2
    $ aws s3 cp --recursive --acl public-read /nix/store/8frm8kk8gzpv31r289ai5jgkwfikmpm4-stdenv-bootstrap-tools-powerpc64-unknown-linux-gnuabielfv2/on-server/ s3://nixpkgs-tarballs/stdenv/powerpc64-unknown-linux-gnuabielfv2/57cf2e0b24fb52344cc718913eaed78f389b1319
    $ aws s3 cp --recursive s3://nixpkgs-tarballs/stdenv/powerpc64-unknown-linux-gnuabielfv2/57cf2e0b24fb52344cc718913eaed78f389b1319 ./
    $ sha256sum bootstrap-tools.tar.xz busybox
    $ sha256sum /nix/store/8frm8kk8gzpv31r289ai5jgkwfikmpm4-stdenv-bootstrap-tools-powerpc64-unknown-linux-gnuabielfv2/on-server/*
```

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] powerpc64-linux (`stdenv` and `hello` builds)
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Artturin** [comment]
It must also be added to `pkgs/stdenv/linux/make-bootstrap-tools-cross.nix`

**Comment by Artturin** [comment]
Also add it to make-bootstrap-tools-cross.nix if https://github.com/NixOS/nixpkgs/pull/258476 is merged before this

**Comment by OPNA2608** [comment]
> Also readd it to make-bootstrap-tools-cross.nix if https://github.com/NixOS/nixpkgs/pull/258476 is merged before this

It was merged before this, so: Rebased on a post-#258476 commit, re-added ppc64 to make-bootstrap-tools-cross.nix.

File: pkgs/stdenv/linux/make-bootstrap-tools-cross.nix [hunk 1]
```diff
@@ -22,6 +22,7 @@ in lib.mapAttrs (n: make) (with lib.systems.examples; {
   mipsel-linux-gnu         = mipsel-linux-gnu;
   mips64el-linux-gnuabin32 = mips64el-linux-gnuabin32;
   mips64el-linux-gnuabi64  = mips64el-linux-gnuabi64;
+  powerpc64 = ppc64;
```
Comment (Artturin) [inline pkgs/stdenv/linux/make-bootstrap-tools-cross.nix:25]: Also a another one https://github.com/NixOS/nixpkgs/pull/258540

(working on automating the bootstrap file updates, that's why it's changing so much)

Comment (OPNA2608) [reply pkgs/stdenv/linux/make-bootstrap-tools-cross.nix:25]: I could change it to `powerpc64-linux-gnuabielfv2`, or I think I wait for the exact formatting of these to be decided over in that PR.

Comment (OPNA2608) [reply pkgs/stdenv/linux/make-bootstrap-tools-cross.nix]: Matched with the changes in https://github.com/NixOS/nixpkgs/pull/258540.

**Comment by OPNA2608** [comment]
Resolved merge conflict after https://github.com/NixOS/nixpkgs/pull/258540 merge.

**Comment by OPNA2608** [comment]
I guess let's try to get the cross bootstrapTools for powerpc64-linux back onto Hydra first, to show that / check if they still work…

**Comment by OPNA2608** [comment]
https://github.com/NixOS/nixpkgs/blob/e0d20bfe6c1f4fc3eb2e20799133fb5d2256ad9d/pkgs/stdenv/linux/make-bootstrap-tools-cross.nix#L15

Ah, well, or not. I guess the convention of referencing the Hydra builds that `bootstrap-files` isn't relevant anymore…

I'll just try to generate some fresher builds then, been afew months.

**Comment by OPNA2608** [comment]
Regenerated, and uploaded… This time just into the tree, with a follow-up that fetches them from the previous commit. Has to be good enough until they go onto the tarballs server I guess :face_in_clouds:.

Some first few derivations in the native bootstrapping chain built fine so far: [bootstrap.log](https://github.com/NixOS/nixpkgs/files/13929822/bootstrap.log)

**Comment by OPNA2608** [comment]
```
checking for references to /build/ in /nix/store/fnl8ksa55r53nr6rp6hnlf1xwkvzn15x-hello-2.12.1...
gzipping man pages under /nix/store/fnl8ksa55r53nr6rp6hnlf1xwkvzn15x-hello-2.12.1/share/man/
patching script interpreter paths in /nix/store/fnl8ksa55r53nr6rp6hnlf1xwkvzn15x-hello-2.12.1
stripping (with command strip and flags -S -p) in  /nix/store/fnl8ksa55r53nr6rp6hnlf1xwkvzn15x-hello-2.12.1/bin
/nix/store/fnl8ksa55r53nr6rp6hnlf1xwkvzn15x-hello-2.12.1
puna@HC ~/d/nixpkgs (bootstrap/powerpc64-linux_v2)> /nix/store/fnl8ksa55r53nr6rp6hnlf1xwkvzn15x-hello-2.12.1/bin/hello 
Hello, world!
```

Files seem to work fine, ready for review (with the caveat of the files still having to be put onto tarballs.nixos.org & fetched from there of course).

**Review by trofi — APPROVED** [review]

**Comment by trofi** [comment]
We'll need to upload tarballs to `s3` as well. /cc @lovesegfault

**Comment by lovesegfault** [comment]
Uploaded!

```
$ aws s3 cp --recursive s3://nixpkgs-tarballs/stdenv/powerpc64-unknown-linux-gnuabielfv2/57cf2e0b24fb52344cc718913eaed78f389b1319 ./
download: s3://nixpkgs-tarballs/stdenv/powerpc64-unknown-linux-gnuabielfv2/57cf2e0b24fb52344cc718913eaed78f389b1319/busybox to ./busybox
download: s3://nixpkgs-tarballs/stdenv/powerpc64-unknown-linux-gnuabielfv2/57cf2e0b24fb52344cc718913eaed78f389b1319/bootstrap-tools.tar.xz to ./bootstrap-tools.tar.xz

$ sha256sum bootstrap-tools.tar.xz busybox
0af311476b54b399f3024b92e9c518363acd2b15f713e83d0bb4fb3f8f26d98b  bootstrap-tools.tar.xz
529a053a8022e89357aaa608e57aaddaa7c3ded93d633916ddca92bb81e22125  busybox
```

**Pull request merged**